code/controllers/subsystem/language.dm

```diff
@@ -4,15 +4,17 @@ SUBSYSTEM_DEF(language)
 	flags = SS_NO_FIRE
 
 /datum/controller/subsystem/language/Initialize()
-	for(var/L in subtypesof(/datum/language))
-		var/datum/language/language = L
-		if(!initial(language.key))
-			continue
+	// ORBSTATION EDIT START
+	if(!GLOB.all_languages.len)
+		for(var/L in subtypesof(/datum/language))
+			var/datum/language/language = L
+			if(!initial(language.key))
```

Comment:
```suggestion
		for(var/datum/language/language_type as anything in subtypesof(/datum/language))
			if(!initial(language_type .key))
```
And so on

Reply:
Actually, this doesn't work, because it's specifically retrieving _typepaths_, which are not a language datum. As for why it's assigning them as a language datum immediately after anyway... I don't have the foggiest idea. I've put it back for now, but this might need some scrutiny upstream.

Reply:
That's what `as anything` is for, you're typecasting every variable in the list to a `datum/language/language_type` even if it... actually isn't one, which saves you from doing that on the immediate next line.